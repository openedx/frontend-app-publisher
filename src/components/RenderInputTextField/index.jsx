import React from 'react';
import PropTypes from 'prop-types';
import { Form } from '@openedx/paragon';

const RenderInputTextField = ({
  input,
  extraInput,
  name,
  label,
  type,
  disabled,
  required,
  maxLength,
  placeholder,
  pattern,
  meta: { touched, error },
}) => (
  <Form.Group controlId={`${input.name || name}-text-label`} isInvalid={touched && error}>
    <Form.Label>
      {label}
    </Form.Label>
    <Form.Control
      {...input}
      {...extraInput}
      id={`${name}-text-label`}
      placeholder={placeholder}
      maxLength={maxLength}
      pattern={pattern}
      name={name}
      type={type}
      disabled={disabled}
      required={required}
    />
    {touched && error && (
    <Form.Control.Feedback>
      {error}
    </Form.Control.Feedback>
    )}
  </Form.Group>
);

RenderInputTextField.defaultProps = {
  extraInput: {},
  name: '',
  disabled: false,
  required: false,
  maxLength: '',
  placeholder: '',
  pattern: null,
};

RenderInputTextField.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string,
  }).isRequired,
  extraInput: PropTypes.shape({}),
  name: PropTypes.string,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  maxLength: PropTypes.string,
  placeholder: PropTypes.string,
  pattern: PropTypes.string,
};

export default RenderInputTextField;
