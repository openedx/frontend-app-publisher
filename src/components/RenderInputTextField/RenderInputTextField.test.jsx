import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RenderInputTextField from './index';

describe('RenderInputTextField', () => {
  it('renders html for text', () => {
    const component = shallow(<RenderInputTextField
      input={{}}
      type="text"
      name="Test"
      label="TestLabel"
      meta={{ touched: false, error: '' }}
    />).dive();
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html for number type', () => {
    const component = shallow(<RenderInputTextField
      input={{}}
      type="number"
      name="Test"
      label="TestLabel"
      meta={{ touched: false, error: '' }}
    />).dive();
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
