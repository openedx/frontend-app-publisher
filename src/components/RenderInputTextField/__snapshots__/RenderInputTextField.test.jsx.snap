// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderInputTextField renders html for number type 1`] = `
<asInput(withDeprecatedProps(InputText))
  dangerIconDescription=""
  disabled={false}
  inline={false}
  isValid={true}
  label="TestLabel"
  maxLength=""
  name="Test"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyPress={[Function]}
  pattern={null}
  placeholder=""
  required={false}
  themes={
    Array [
      "danger",
    ]
  }
  type="number"
  validationMessage=""
  value=""
/>
`;

exports[`RenderInputTextField renders html for text 1`] = `
<asInput(withDeprecatedProps(InputText))
  dangerIconDescription=""
  disabled={false}
  inline={false}
  isValid={true}
  label="TestLabel"
  maxLength=""
  name="Test"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyPress={[Function]}
  pattern={null}
  placeholder=""
  required={false}
  themes={
    Array [
      "danger",
    ]
  }
  type="text"
  validationMessage=""
  value=""
/>
`;
